GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -768,6 +792,31 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
             return result;
         }
 
+        private FileSystemTaskResult TryRemoveModifiedPath(string virtualPath, bool isFolder)
+        {
+            if (!this.newlyCreatedFileAndFolderPaths.Contains(virtualPath))
+            {
+                return FileSystemTaskResult.Invalid;
+            }

```

Comment:
For what it's worth, I think it was OK when you had the `newlyCreatedFileAndFolderPaths` check outside of `TryRemoveModifiedPath`.

When I suggested adding a helper I wasn't sure what you'd think of something like:

```
private FileSystemTaskResult TryUpdateModifiedPathsForRemoval(string virtualPath, bool isFolder)
{
    if (this.newlyCreatedFileAndFolderPaths.Contains(virtualPath))
    {
        return this.TryRemoveModifiedPath(virtualPath, isFolder);
    }

    return this.TryAddModifiedPath(virtualPath, isFolder);
}
```

And then calling that method in the three spots that needs that logic (up to you which you like better, I'm just throwing this out there 😃)

Reply:
I thought about it but couldn't come up with a good name for the method that wasn't `TryRemoveFromModifiedPathsIfNewlyCreatedOrAddIfNotNewlyCreated` :)